codewof/static/js/question_types/debugging.js

```diff
@@ -152,3 +158,56 @@ function run_python_code(user_code, test_case) {
         test_case.runtime_error = true;
     }
 }
+
+
+function setTutorialAttributes() {
+    $(".question-text").attr(
+        'data-intro',
+        'This is a description of what the code should do. In debugging questions there will be one or more bugs in the given code that you have to find and correct in order for all the tests to pass.'
+    );
+    $("#python-editor").attr(
+        'data-intro',
+        "This is the code that contains the bug(s). Find the bug(s) and correct them. The greyed out lines cannot be edited."
+    );
+    $("#run_code").attr(
+        'data-intro',
+        "Click 'Run code' to run your code against the test cases and see if you fixed the bug!"
+    );
+    $("#reset_to_initial").attr(
+        'data-intro',
+        "You can click 'Reset to initial code' at any time to discard all of your changes."
+    );
+    $("#test-case-table").attr(
+        'data-intro',
+        "These are the test cases that have been run against the given code."
+    );
+    // the first row in the test case table
+    $('#test-case-table tbody tr:nth-child(1)').attr(
+        'data-intro',
+        'Here is the first test case.'
+    );
+    // the input for the first test case
+    $('#test-case-table tbody tr:nth-child(1) td:eq(0)').attr(
+        'data-intro',
+        'This is the test code that has been run for this particular test.'
+    );
+    // the expected output for the first test case
+    $('#test-case-table tbody tr:nth-child(1) td:eq(1)').attr(
+        'data-intro',
+        'This is the output that the test case expects to be printed for the given input.'
+    );
+    // the received output for the first test case
+    $('#test-case-table tbody tr:nth-child(1) td:eq(2)').attr(
+        'data-intro',
+        'This is the output that has been printed by the test case.\
+        The test case either prints the return value of the code, or the code executed by the test prints it itself.'
+    );
+    // the status of the first test case
+    $('#test-case-table tbody tr:nth-child(1) td:eq(3)').attr(
+        'data-intro',
+        "This is the status of the test. It will say one of three things: 'Not yet run', 'Passed' or 'Failed'.\
+        'Not yet run' means your code has not yet been run against the test case.\
+        'Passed' means the received output matched the expected output.\
+        'Failed' means the received output did not match the expected output."
```

Comment:
I still think it could be as simple as "A test case will pass if the received output matches the expected output. If all test cases pass the question has been solved." or something similar. Those three terms are self-explanatory in this context